sky/serve/autoscalers.py

```diff
@@ -16,32 +17,36 @@
 
 logger = sky_logging.init_logger(__name__)
 
-_UPSCALE_DELAY_S = 300
-_DOWNSCALE_DELAY_S = 1200
-
 
 class AutoscalerDecisionOperator(enum.Enum):
     SCALE_UP = 'scale_up'
     SCALE_DOWN = 'scale_down'
-    NO_OP = 'no_op'
 
 
 @dataclasses.dataclass
 class AutoscalerDecision:
     """Autoscaling decisions.
 
-    |---------------------------------------------------------|
-    | Operator   | TargetType | Meaning                       |
-    |------------|------------|-------------------------------|
-    | SCALE_UP   | int        | Number of replicas to add     |
-    |------------|------------|-------------------------------|
-    | SCALE_DOWN | List[int]  | List of replica ids to remove |
-    |------------|------------|-------------------------------|
-    | NO_OP      | None       | No scaling needed             |
-    |---------------------------------------------------------|
+    |---------------------------------------------------------------|
+    | Operator   | TargetType       | Meaning                       |
+    |------------|------------------|-------------------------------|
+    | SCALE_UP   | Dict[str, Any]   | Resource override to add      |
+    |------------|------------------|-------------------------------|
+    | SCALE_DOWN | int              | Replica id to remove          |
+    |---------------------------------------------------------------|
     """
     operator: AutoscalerDecisionOperator
-    target: Optional[Union[int, List[int]]]
+    target: Optional[Union[Dict[str, Any], int]]
+
+    def __init__(self,
+                 operator: AutoscalerDecisionOperator,
+                 target: Optional[Union[Dict[str, Any], int]] = None):
+
+        # Scale down requires replica ids to remove.
+        assert not (operator == AutoscalerDecisionOperator.SCALE_DOWN and
+                    target is None)
+        self.operator = operator
+        self.target = target
```

Comment:
After previous mentioned type change, we should be able to remove this function

Reply:
with `Union[Optional[Dict[str, Any], int]]`,
you can still specify `AutoscalerDecisionOperator.SCALE_DOWN and
                    target is None`